src/hooks/usePodBrowserSettings/index.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { useContext, useEffect, useState } from "react";
+import { getSolidDataset } from "@inrupt/solid-client";
+import useAuthenticatedProfile from "../useAuthenticatedProfile";
+import { getOrCreateSettings } from "../../solidClientHelpers/settings";
+import SessionProvider from "../../contexts/sessionContext";
+
+export default function usePodBrowserSettings() {
+  const { session } = useContext(SessionProvider);
+  const profileInfo = useAuthenticatedProfile();
+  const [settings, setSettings] = useState(null);
+
+  useEffect(() => {
+    if (!profileInfo || !session) return;
+    getOrCreateSettings(profileInfo.webId, session)
+      .then((settingsUrl) =>
+        getSolidDataset(settingsUrl, { fetch: session.fetch })
+      )
+      .then(setSettings);
```

Comment:
Imo, this would read more easily as an async:

```
useEffect(() => {
  (async function getSettings() {
    const settingsUrl = await getOrCreateSettings(profileInfo.webId, session);
    const settings = await getSolidDataset(settingsUrl, { fetch: session.fetch });
    setSettings(settings);
  })();
});
```

Reply:
I agree, just didn't think we wanted to have async/await in the useEffect block.

Will change it.

Reply:
It's situational - I tend not to if there's a single promise, but if there are multiple, I think it's easier to read.

Reply:
After some changes there's only one promise, so I didn't use the async/await then.